### DISCUSSION TITLE
Please provide the references for the g.problem

### DISCUSSION THREAD

**Discussion by zixiangliwust** [anyoptimization/pymoo#275] [Q&A]
Where are the G1, G2,  G3, G4 and so on for the single objective taken from? Please provide the reference.

**Comment by blankjul** [comment]
I had to look up the source. It is now added as a comment to the source file (in the release branch).

Liang, Jing J., Thomas Philip Runarsson, Efrén Mezura-Montes, Maurice Clerc, Ponnuthurai Nagaratnam Suganthan, Carlos A. Coello Coello, and Kalyanmoy Deb.
Problem Definitions and Evaluation Criteria for the CEC 2006 Special Session on Constrained Real-Parameter Optimization.
